library/unistd/spawnvpe.c

```diff
@@ -233,15 +239,14 @@ spawnvpe(
     arg_string[arg_string_len] = '\0';
 
     D(("arg_string: [%s]\n", arg_string));
-    printf("arg_string: [%s]\n", arg_string);
 
-    int finalpath_len = strlen(name) + 1 + arg_string_len + 1; // '\0'
-    char *finalpath = (char*)malloc(finalpath_len);
-    char processName[32] = {0};
-    snprintf(finalpath, finalpath_len, "%s %s", name, arg_string);
-    snprintf(processName, NAMELEN - 1, "Spawned Process #%d", __clib4->__children);
+    int full_command_len = strlen(name) + 1 + arg_string_len + 1; // '\0'
+    char *full_command = (char*)malloc(full_command_len);
```

Comment:
I see that you renamed the `finalpath` to `full_command` but I do not see where this is freed. Maybe that was forgotten or we missed it in first place. Should we free that memory?